src/common/config/env.validation.ts

```diff
@@ -164,6 +168,14 @@ export class EnvironmentVariables {
   @ValidateIf((vars) => vars.ETH_NETWORK === Network.Mainnet)
   public START_EPOCH = 155000;
 
+  @IsInt()
+  @IsPositive()
+  @IsValidDencunEpoch()
+  @Expose()
+  @Transform(transformDencunEpoch)
```

Comment:
Let's make this thing simpler. I would suggest this:
```suggestion
  @Expose()
  @IsNumber({ allowNaN: true })
  @IsPositive()
  @Transform(({ value, obj }) => dencunForkEpoch[obj.ETH_NETWORK] ?? parseInt(value, 10), { toClassOnly: true })
```
1.  It forces the use predefined values for known networks
2. This epoch should never achieved for unknown network by default, so `NaN` value will be valid because `NaN > int`. 

I also suggest logging this value when the app is started

Reply:
Thank you for the detailed analysis of this PR!

If I correctly understand the 2nd suggestion, you think that the app should assume that unknown networks don't support Dencun hardfork by default. So, you suggest allowing users not to set the Dencun hardfork epoch for custom networks. The app should treat absent or invalid values of `DENCUN_FORK_EPOCH` for unknown networks like if Dencun hardfork will never happen for this network. Do I correctly understand your suggestion? If yes, could you please clarify why you think so?

I'm not sure that it is a good assumption. As for me, it's better to force users to set the Dencun hardfork epoch for their custom networks. If a custom network doesn't support Dencun (and doesn't plan to support it), a user should clearly set the value of the `DENCUN_FORK_EPOCH` variable far away in the future. I admit that sometimes later there might be a situation when custom networks already support Dencun, but a user forgot to set `DENCUN_FORK_EPOCH` variable and the E-V-M app will work incorrectly in this case.
@vgorkavenko what do you think?

If we decide that the app should behave in the way that you suggested, I can surely correct the PR to support this behavior, but the approach with `{ allowNaN: true }` will not work anyway, because in JS `NaN > num === false` and `NaN < num === false`.

Reply:
1.  > `NaN > num === false`

Yes, you're right. We can use `Infinity`
`dencunForkEpoch[obj.ETH_NETWORK] ?? (value ? parseInt(value, 10) : Infinity)`

2. It is okay not to force specifying a fork epoch for dev or local networks
Example from CL client: https://github.com/ChainSafe/lodestar/blob/unstable/packages/config/src/chainConfig/networks/ephemery.ts

Reply:
OK, I agree with these your suggestions. I've simplified the validation engine and implemented your suggestion with a few minor tweaks. I've also added logging of the `DENCUN_FORK_EPOCH` value.

It is important to note that now the validation engine is based on these 2 implicit assumptions https://github.com/lidofinance/ethereum-validators-monitoring/pull/229#issuecomment-1884740941

If you agree that the engine should work in accordance with these assumptions, I think we should clearly write them in Readme. I'll update Readme if we both agree with the new implementation.